psl/psl-core/src/common/preview_features.rs

```diff
@@ -85,88 +88,193 @@ features!(
+                    }),
+                ),
+            ]),
+            renamed: BTreeMap::from([
+                #[cfg(feature = "postgresql")]
+                (
+                    RenamedFeatureKey {
+                        from: PreviewFeature::FullTextSearch,
+                        provider: Some("postgresql"),
+                    },
+                    RenamedFeatureValue {
+                        to: PreviewFeature::FullTextSearchPostgres,
+                        prisly_link_endpoint: "fts-postgres",
+                    },
+                ),
+            ]),
+            deprecated: enumflags2::make_bitflags!(PreviewFeature::{
+                AtomicNumberOperations
+                | AggregateApi
+                | ClientExtensions
+                | Cockroachdb
+                | ConnectOrCreate
+                | CreateMany
+                | DataProxy
+                | Distinct
+                | ExtendedIndexes
+                | ExtendedWhereUnique
+                | FieldReference
+                | FilteredRelationCount
+                | FilterJson
+                | FullTextIndex
+                | FullTextSearch
+                | GroupBy
+                | ImprovedQueryRaw
+                | InteractiveTransactions
+                | JsonProtocol
+                | MicrosoftSqlServer
+                | Middlewares
+                | MongoDb
+                | NamedConstraints
+                | NApi
+                | NativeTypes
+                | OrderByAggregateGroup
+                | OrderByNulls
+                | OrderByRelation
+                | ReferentialActions
+                | ReferentialIntegrity
+                | SelectRelationCount
+                | TransactionApi
+                | UncheckedScalarInputs
+            }),
+            hidden: enumflags2::make_bitflags!(PreviewFeature::{ReactNative | TypedSql}),
+        };
+
+        Self {
+            provider: connector_provider,
+            feature_map,
+        }
+    }
+
+    pub fn native_features(&self) -> PreviewFeatures {
+        self.provider
+            .and_then(|provider| self.feature_map.native.get(provider).copied())
+            .unwrap_or_default()
+    }
+
+    pub fn active_features(&self) -> PreviewFeatures {
+        self.feature_map.active | self.native_features()
     }
 
     pub const fn hidden_features(&self) -> PreviewFeatures {
-        self.hidden
+        self.feature_map.hidden
     }
 
     pub(crate) fn is_valid(&self, flag: PreviewFeature) -> bool {
-        (self.active | self.hidden).contains(flag)
+        (self.active_features() | self.feature_map.hidden).contains(flag)
     }
 
     pub(crate) fn is_deprecated(&self, flag: PreviewFeature) -> bool {
-        self.deprecated.contains(flag)
+        self.feature_map.deprecated.contains(flag)
+    }
+
+    /// Was the given preview feature deprecated and renamed?
+    pub(crate) fn is_renamed(&'a self, flag: PreviewFeature) -> Option<RenamedFeature<'a>> {
```

Comment:
This unnecessarily restricts the availability of this method if you have a reference to `FeatureMapWithProvider` valid for shorter than `'a` (i.e. `&'b FeatureMapWithProvider<'a> where 'a: 'b`), just
```suggestion
    pub(crate) fn is_renamed(&self, flag: PreviewFeature) -> Option<RenamedFeature<'_>> {
```
is fine here.

Reply:
Good catch, this is no longer needed.